d3log/src/batch.rs

```diff
@@ -0,0 +1,173 @@
+use differential_datalog::{DeltaMap, program::RelId, program::Update, ddval::DDValue};
+use mm_ddlog::*;
+use serde::{de, de::SeqAccess, de::Visitor, ser::SerializeTuple, Serialize, Deserialize, Serializer, Deserializer};
+use std::convert::TryFrom;
+use std::fmt;
+use std::fmt::Display;
+use std::io::{Error,ErrorKind};
+use std::collections::BTreeMap;
+
+// the other choice here would be a Vec<Update>?
+// need multiplicity
+// #[derive(Serialize, Deserialize)] - deltamap isn't serialize
+pub struct Batch {
+    // one might consider augmenting the values w/ t - kind of a
+    // performance problem for multisets
+    timestamp: u64, 
+    pub b: DeltaMap<differential_datalog::ddval::DDValue>,
+    // timestamp
+}
+
+impl Serialize for Batch {
+    // i would _like_ to expose an interface that used the names for relations
+    // so that external users dont have to be privy to the compiler id assignemnt
+    // xxx - u128 is not supported!
```

Comment:
https://github.com/serde-rs/json/issues/448

looks like this is maybe still open?

Reply:
umm...I just changed this and its not throwing an error...no idea why I was having problems earlier. there _is_ an issue that according to json we can only send 64 bit floats, so depending on the actual implementation  - this may not make it to the other side intact

Reply:
verified that this runs - I could swear .. never mind